### ISSUE TITLE
Location for (current version of) XML Schema for XML configuration file has changed

### ISSUE THREAD

**Issue by morozov** [sebastianbergmann/phpunit#4405]
Up until PHPUnit 9.3, it was possible to specify the configuration schema as the following:
```xml
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
```
This way, after a PHPUnit update, it was possible to validate the configuration against the current schema.

As of PHPUnit 9.3, the schema path has to be specified like this:
```xml
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/schema/9.3.xsd"
```
Which requires an additional non-obvious code change during PHPUnit updates.

It would be nice if the following was possible:
```xml
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/schema/current.xsd"
```

**Comment by sebastianbergmann** [comment]
I did not anticipate anybody to use a relative path such as `xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"`. In all projects except for PHPUnit itself, I always use `xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"`, for instance.

I would like to avoid a special case for the filename of the current schema as this would make the code that deals with different schema versions less consistent / more complicated than it needs to be.

**Comment by morozov** [comment]
All I'm suggesting is the alias (e.g. a hard copy of or a symlink to the current schema). The code that needs to be aware of multiple versions will use the specific versions. The code that needs the current version will refer to the current version.

The usage of a URL or path that contains a specific version requires additional and easy-to-forget maintenance during PHPUnit upgrades.

**Comment by morozov** [comment]
> I would like to avoid a special case for the filename of the current schema as this would make the code that deals with different schema versions less consistent / more complicated than it needs to be.

IMO, the usage of different schema versions is the special case currently being optimized for. The usage of the current version is a more common case.

**Comment by remorhaz** [comment]
Using "current" scheme just breaks the link between the actual file grammar and the scheme to check it. I believe keeping the scheme versioned is absolutely good decision: it allows user to get a notification when the grammar changes (and that happened in 9.3, BTW).

**Comment by morozov** [comment]
> it allows user to get a notification when the grammar changes

But isn't it what PHPUnit does internally by using the specific schema version? What I'm talking about is the usage in IDEs where it validates the configuration against the schema. If the "current" approach was used, the user would see that they are using a no longer valid schema even before running PHPUnit.

**Comment by morozov** [comment]
Another use case where hard-coding the schema version is annoying is automated dependency updates via @dependabot-preview. It will update the PHPUnit version, and if all tests pass, the maintainer may merge the PR w/o even noticing that the schema used in the configuration no longer exists (and that happened in 9.3, BTW).

**Comment by sebastianbergmann** [comment]
> I did not anticipate anybody to use a relative path such as `xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"`.

To clarify what I meant by that: I consider the file that used to be `/phpunit.xsd` and now is `/schema/9.3.xsd` a private implementation detail of PHPUnit. The public API, if you will, is the `https://schema.phpunit.de/9.3/phpunit.xsd` URL.

**Comment by morozov** [comment]
> To clarify what I meant by that: I consider the file that used to be `/phpunit.xsd` and now is `/schema/9.3.xsd` a private implementation detail of PHPUnit. The public API, if you will, is the `https://schema.phpunit.de/9.3/phpunit.xsd` URL.

This justifies the non-BC-breaking nature of this change which isn't a concern here. The concern is the absence of the public API for the current schema version.

**Comment by morozov** [comment]
@sebastianbergmann given the logic implemented in `SchemaFinder::find()` by e2b5eeac543b699867a85391e2bfe039450b881c, is it now safe to rely on `schema/phpunit.xsd` being part of the PHPUnit library or it's still an implementation detail?

**Comment by ben221199** [comment]
Seems to me that the Laravel Installer should use that absolute namespace too.
